### DISCUSSION TITLE
scaling containers/pods

### DISCUSSION THREAD

**Discussion by tcurdt** [containers/podman#20525] [Q&A]
Let's say I want to adjust the amount of (backend) container instances on a single node.
There is no way with podman to let me easily adjust that, is there?
It seems the pod command does not have "scale" or similar.

**Comment by eriksjolund** [comment]
The closest thing I can think of is using [_socket activation of containers_](https://github.com/containers/podman/blob/main/docs/tutorials/socket_activation.md?ref=xzllll.com#socket-activation-of-containers) togehter with [`ReusePort=true`](https://www.freedesktop.org/software/systemd/man/latest/systemd.socket.html#ReusePort=) and [quadlet templates]( https://docs.podman.io/en/latest/markdown/podman-systemd.unit.5.html#template-files).


I wrote an [example](https://github.com/eriksjolund/podman-networking-docs?tab=readme-ov-file#example-use-reuseporttrue-to-share-a-port-between-services) where I can scale up to 2 by running
```
systemctl --user start app@1.socket
systemctl --user start app@2.socket
```

"1" and "2" are just a text strings in the example. The socket unit _app@1.socket_ activates _app@1.service_ and _app@2.socket_ activates _app@2.service_ . When a client connects to the port, the linux kernel will distribute the incoming connection to the socket in one the two services.
